src/Catalog.Api/Mappers/CategoryMapper.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Catalog.Api.Requests;
+using Catalog.Api.Responses;
+
+namespace Catalog.Api.Mappers
+{
+    public static class CategoryMapper
+    {
+        public static Domain.Entities.Category Map(CategoryRequest request, int? categoryId = null)
+        {
+            return new Domain.Entities.Category
+            {
+                Id = categoryId ?? 0,
+                Name = request.Name,
+                ImageUrl = request.ImageUrl,
+                Parent = request.ParentCategoryId != null ? new Domain.Entities.Category
+                {
+                    Id = (int)request.ParentCategoryId
```

Comment:
nullables has `Value` property, consider using that instead of casting nullable to `int` as in case parent category is not provided, everything will explode

Reply:
Fixed